xapian-core/tests/api_db.cc

```diff
@@ -1705,63 +1745,28 @@ DEFINE_TESTCASE(netstats1, remote) {
     rset.add_document(4);
     rset.add_document(9);
 
-    Xapian::MSet mset_alllocal;
-    {
-	Xapian::Database db;
-	db.add_database(local_manager.get_database("apitest_simpledata"));
-	db.add_database(local_manager.get_database("apitest_simpledata2"));
-
-	Xapian::Enquire enq(db);
-	enq.set_query(query);
-	mset_alllocal = enq.get_mset(0, MSET_SIZE, &rset);
-    }
-
-    {
-	Xapian::Database db;
-	db.add_database(local_manager.get_database("apitest_simpledata"));
-	db.add_database(get_database("apitest_simpledata2"));
-
-	Xapian::Enquire enq(db);
-	enq.set_query(query);
-	Xapian::MSet mset = enq.get_mset(0, MSET_SIZE, &rset);
-	TEST_EQUAL(mset.get_matches_lower_bound(), mset_alllocal.get_matches_lower_bound());
-	TEST_EQUAL(mset.get_matches_upper_bound(), mset_alllocal.get_matches_upper_bound());
-	TEST_EQUAL(mset.get_matches_estimated(), mset_alllocal.get_matches_estimated());
-	TEST_EQUAL(mset.get_max_attained(), mset_alllocal.get_max_attained());
-	TEST_EQUAL(mset.size(), mset_alllocal.size());
-	TEST(mset_range_is_same(mset, 0, mset_alllocal, 0, mset.size()));
-    }
-
-    {
-	Xapian::Database db;
-	db.add_database(get_database("apitest_simpledata"));
-	db.add_database(local_manager.get_database("apitest_simpledata2"));
-
-	Xapian::Enquire enq(db);
-	enq.set_query(query);
-	Xapian::MSet mset = enq.get_mset(0, MSET_SIZE, &rset);
-	TEST_EQUAL(mset.get_matches_lower_bound(), mset_alllocal.get_matches_lower_bound());
-	TEST_EQUAL(mset.get_matches_upper_bound(), mset_alllocal.get_matches_upper_bound());
-	TEST_EQUAL(mset.get_matches_estimated(), mset_alllocal.get_matches_estimated());
-	TEST_EQUAL(mset.get_max_attained(), mset_alllocal.get_max_attained());
-	TEST_EQUAL(mset.size(), mset_alllocal.size());
-	TEST(mset_range_is_same(mset, 0, mset_alllocal, 0, mset.size()));
-    }
-
     {
-	Xapian::Database db;
-	db.add_database(get_database("apitest_simpledata"));
-	db.add_database(get_database("apitest_simpledata2"));
+	Xapian::Database db = get_database("netstats1", make_netstats1_db);
 
 	Xapian::Enquire enq(db);
 	enq.set_query(query);
 	Xapian::MSet mset = enq.get_mset(0, MSET_SIZE, &rset);
-	TEST_EQUAL(mset.get_matches_lower_bound(), mset_alllocal.get_matches_lower_bound());
-	TEST_EQUAL(mset.get_matches_upper_bound(), mset_alllocal.get_matches_upper_bound());
-	TEST_EQUAL(mset.get_matches_estimated(), mset_alllocal.get_matches_estimated());
-	TEST_EQUAL(mset.get_max_attained(), mset_alllocal.get_max_attained());
-	TEST_EQUAL(mset.size(), mset_alllocal.size());
-	TEST(mset_range_is_same(mset, 0, mset_alllocal, 0, mset.size()));
+	TEST_EQUAL(mset.get_matches_lower_bound(), 7);
+	TEST_EQUAL(mset.get_matches_upper_bound(), 7);
+	TEST_EQUAL(mset.get_matches_estimated(), 7);
+	TEST_EQUAL(mset.get_max_attained(), 1.445962071042388164);
+	TEST_EQUAL(mset.size(), 7);
+
+	vector<pair<Xapian::docid, double>> to_compare = {
```

Comment:
We don't need to be able to modify this, so better to make it an array and `static const` - i.e. `static const pair<Xapian::docid, double> to_compare[]` so the compiler can put it in the read-only data section at compile time.